src/validations/access-checker/items/third-party-asset.ts

```diff
@@ -0,0 +1,137 @@
+import { verifyProof } from '@dcl/content-hash-tree'
+import { keccak256Hash } from '@dcl/hashing'
+import { isThirdParty, MerkleProof, ThirdPartyProps } from '@dcl/schemas'
+import { BlockchainCollectionThirdParty } from '@dcl/urn-resolver'
+import { ILoggerComponent } from '@well-known-components/interfaces'
+import { ContentValidatorComponents, DeploymentToValidate, OK, validationFailed } from '../../../types'
+import { AssetValidation } from './items'
+
+function toHexBuffer(value: string): Buffer {
+  if (value.startsWith('0x')) {
+    return Buffer.from(value.substring(2), 'hex') // removing first 2 characters (0x)
+  }
+  return Buffer.from(value, 'hex')
+}
+
+function getThirdPartyId(urn: BlockchainCollectionThirdParty): string {
+  return `urn:decentraland:${urn.network}:collections-thirdparty:${urn.thirdPartyName}`
+}
+
+async function verifyHash<T>(
+  metadata: T & ThirdPartyProps,
+  merkleRoot: string,
+  logger: ILoggerComponent.ILogger
+): Promise<boolean> {
+  if (!MerkleProof.validate(metadata.merkleProof)) {
+    logger.debug('Merkle proof is not valid', {
+      merkleProof: metadata.merkleProof
+    })
+    return false
+  }
+  const merkleProof = metadata.merkleProof
+  const generatedCrcHash = keccak256Hash(metadata, metadata.merkleProof.hashingKeys)
+  // The hash provided in the merkleProof for the entity MUST match the hash generated by the validator.
+  if (metadata.merkleProof.entityHash !== generatedCrcHash) {
+    logger.debug(`The hash provided in the merkleProof doesn't match with the generated by the validator`, {
+      generatedCrcHash,
+      entityHash: metadata.merkleProof.entityHash
+    })
+    return false
+  }
+
+  // Verify if the entity belongs to the Merkle Tree.
+  const bufferedProofs = merkleProof.proof.map((value) => toHexBuffer(value))
+  const bufferedMerkleRoot = toHexBuffer(merkleRoot)
+  return verifyProof(merkleProof.index, merkleProof.entityHash, bufferedProofs, bufferedMerkleRoot)
+}
+
+async function getMerkleRoot(
+  components: Pick<ContentValidatorComponents, 'externalCalls'>,
+  block: number,
+  thirdPartyId: string
+): Promise<string | undefined> {
+  const query = `
```

Comment:
This is what I mentioned earlier... I think it should be nicer to add such a query to TheGraphClient and then receive it here (instead of externalCalls).

Reply:
Yup, i `completely` agree. Just wanted to put an end to the PR but definitely should use the graph client.